vshard/replicaset.lua

```diff
@@ -697,10 +699,17 @@ local function replica_call(replica, func, args, opts)
            err.type == 'ClientError') then
             err = lerror.from_string(err.message) or err
         end
-        log.error("Exception during calling '%s' on '%s': %s", func, replica,
-                  err)
+        local log_level = lerror.are_same(err, replica.last_error) and
+                          'verbose' or 'error'
+        log[log_level]("Exception during calling '%s' on '%s': %s",
+                       func, replica, err)
+        replica.last_error = err
```

Comment:
The problem here is that if the errors are of different types or contain the called function name or are different in any other way, and are being all spammed together, then there will still be spam, because `last_error` will not be the same. It would only work if there would be one error happening repeatedly with no other errors at all.

This also forces us to check and nullify the `last_error` on the success path, which adds one more `if`.